src/upf_network.py

```diff
@@ -49,12 +51,70 @@ def get_ip_address(self) -> str:
             ip_addresses = iface_record.ipaddr
             for ip in ip_addresses:
                 if ip.family == AF_INET:
-                    return ip.address
+                    return f"{ip.address}/{ip.prefixlen}"
             logger.warning("No IPv4 address found for interface %s", self.name)
         except KeyError:
             logger.warning("Interface %s not found in the network database", self.name)
         return ""
 
+    def addresses_are_set(self) -> bool:
+        """Check if the given network interface is already configured."""
+        interfaces = self.network_db.interfaces  # type: ignore[reportAttributeAccessIssue]
+        try:
+            iface_record = interfaces[self.name]
+            ip_addresses = iface_record.ipaddr
+            for ip in ip_addresses:
+                if ip.family == AF_INET:
+                    logger.info("Found IP: %s/%s", ip.address, ip.prefixlen)
+                    if f"{ip.address}/{ip.prefixlen}" != self.ip_address:
+                        return False
+            if not self.get_ip_address():
+                return False
+        except KeyError:
+            logger.warning("Interface %s not found in the network database", self.name)
+            return False
+        return True
+
+    def set_ip_address(self) -> None:
+        """Clean all unrequired IPs and set the IP address for the given network interface."""
+        interfaces = self.network_db.interfaces  # type: ignore[reportAttributeAccessIssue]
+        try:
+            iface_record = interfaces[self.name]
+            ip_addresses = iface_record.ipaddr
+            # remove all unrequired IPs
+            for ip in ip_addresses:
+                if ip.family == AF_INET:
+                    if ip.address != self.ip_address:
+                        logger.info("Removing IP %s/%s from interface %s",
+                                    ip.address, ip.prefixlen, self.name)
+                        iface_record.del_ip(f"{ip.address}/{ip.prefixlen}").commit()
+            # add requested IP if not already there
+            if not self.get_ip_address():
+                logger.info("Adding IP %s to interface %s", self.ip_address, self.name)
+                iface_record.add_ip(self.ip_address).commit()
```

Comment:
Does all of this code need to be in the try/except for the key error? Let's be precise when we want to catch errors. This applies in other places as well